src/transformers/models/llama/modeling_llama.py

```diff
@@ -491,6 +491,8 @@ def forward(
 
         past_key_value = (key_states, value_states) if use_cache else None
 
+        # TODO: These transpose are quite inefficient but Flash Attention requires the layout [batch_size, sequence_length, num_heads, head_dim]. We would need to refactor the KV cache
+        # to be able to avoid many of these transpose/reshape/view.
```

Comment:
indeed and can't wait to see the performance gains we can have

Reply:
moved to https://github.com/huggingface/transformers/pull/27625